great_expectations/experimental/metric_repository/cloud_data_store.py

```diff
@@ -11,11 +13,41 @@
 
 
 class CloudDataStore(DataStore[StorableTypes]):
+    def __init__(self, context: AbstractDataContext):
+        super().__init__(context=context)
+        config = self._context._cloud_config
+        self._session = create_session(access_token=config.access_token)
+        self._url = (
+            f"{config.base_url}/organizations/{config.organization_id}/metric-runs"
+        )
+
+    def _construct_json_payload(self, metric_run: MetricRun):
+        result = {
+            "id": str(metric_run.id),
+            "metrics": [],
+        }
+        for metric in metric_run.metrics:
+            result["metrics"].append(
+                {
+                    "id": str(metric.id),
+                    "batch_id": metric.batch.id,
+                    "data_asset_id": str(metric.batch.data_asset.id),
+                    "metric_name": metric.metric_name,
+                    "exception_message": metric.exception.exception_message,
+                    "exception_type": metric.exception.exception_type,
+                    "value": metric.value,
+                    "metric_type": type(metric).__name__,
+                    "value_type": type(metric.value).__name__,
+                    "value_parameter_names": [],
+                }
+            )
+        return {"data": {"attributes": result}}
+
     def add(self, value: T) -> T:
```

Comment:
Currently the method signature here is meant to take in any type and add it (e.g. not just MetricRun). In that case maybe we look up a serializer based on the thing being serialized (metric run) rather than the cloud data store knowing how to serialize a MetricRun. That might be overkill for now. Maybe instead we can rename the methods in here to `add_metric_run` and `_construct_metric_run_json_payload()` so we can make it more generic without breaking the callers later. What do you think?

Reply:
I was thinking about checking the type of T and calling the _construct if it's a MetricRun and just saving the organization url in __init__

Reply:
I also thought about returning the response.json() of the POST instead of the value object that was passed to it.

Reply:
If we check the type of T then each time we add a new data type (besides MetricRun) we will have to modify the CloudDataStore.

Reply:
What about using something like model.model_dump_json(...) https://docs.pydantic.dev/latest/usage/serialization/#modelmodel_dump_json to serialize to json? So we can hand off the serialization responsibilities to pydantic and then as long as we get a pydantic model as the value (one of the StorableTypes, mypy will help us check those) then we can just call the model_dump_json() method.